release.py

```diff
@@ -0,0 +1,694 @@
+            logger.info('Latest tag is not in format X.Y.Z.')
+        else:
+            suggested_tag = bump_minor_version(latest_tag)
+            logger.info(f'Suggested new version: {suggested_tag}')
+
+    tag_prompt = f' (press enter to use the tag {suggested_tag})' if suggested_tag else ''
+    prompt = f"Input the new tag for the release{tag_prompt}, or 'c' to cancel:\n"
+    while True:
+        user_input = input(prompt).strip()
+        if user_input.lower() == 'c':
+            logger.warning('Release canceled.')
+            exit(0)
+
+        new_tag = user_input if user_input else suggested_tag
+
+        if not new_tag:
+            logger.error('Error: No tag specified and no suggestion available')
+            continue
+
+        if not re.match(r'^\d+\.\d+\.\d+$', new_tag):
+            logger.error('Error: Tag must be in format X.Y.Z')
+            continue
+
+        release_page = f'https://github.com/{owner}/{repo.name}/releases'
+        logger.warning(f'Check out the releases page: {release_page} before confirming!')
+
+        confirm = (
+            input(f"Confirm creating tag '{new_tag}' on branch '{branch_name}'? [y/N]: ")
+            .strip()
+            .lower()
+        )
+
+        if confirm == 'y':
+            break
+
+        logger.info("Let's try again...")
+
+    return new_tag
+
+
+def create_draft_release(
+    repo: github.Repository.Repository, tag: str, branch: str
+) -> github.GitRelease.GitRelease | None:
+    """Create a draft release with auto-generated notes."""
+    try:
+        release = repo.create_git_release(
+            tag=tag,
+            name=tag,
+            draft=True,
+            generate_release_notes=True,
+            target_commitish=branch,
+        )
+        return release
+    except Exception as e:
+        logger.error(f'Failed to create release: {e}')
+        return None
+
+
+def parse_release_notes(release_notes: str) -> tuple[dict[str, list[tuple[str, str]]], str | None]:
+    """Parse auto-generated release notes into categories.
+
+    The "New Contributors" section is removed.
+    The PRs are categorized into 'feat', 'fix', 'docs', 'test', and 'ci'.
+    The full changelog line is returned separately.
+
+    Returns:
+        A tuple containing:
+        - A dict with conventional commit types as keys and lists of tuples (description, PR link)
+          as values.
+        - The full changelog line if present, or None if not found.
+    """
+    release_notes = re.sub(
+        r'(## New Contributors.*?)(\n|$)', r'\2', release_notes, flags=re.DOTALL
+    )
+    categories: dict[str, list[tuple[str, str]]] = {
+        'feat': [],
+        'fix': [],
+        'docs': [],
+        'test': [],
+        'ci': [],
+    }
+    full_changelog_line = None
+
+    for line in release_notes.splitlines():
+        if match := re.match(r'^\* (\w+): (.*) by @\w+ in (.*)', line.strip()):
+            category = match.group(1)
+            if category in categories:
```

Comment:
Should we have an `else` and print a log message if we run into a category we don't expect? Or is that not an issue?